package.json

```diff
@@ -105,13 +105,17 @@
 		"rules": {
 			"no-alert": "off",
 			"import/no-unassigned-import": "off",
-			"import/no-unresolved": "warn"
+			"import/no-unresolved": "warn",
+			"import/named": "warn"
 		},
 		"overrides": [
 			{
 				"files": "**/*.+(ts|tsx)",
 				"extends": "xo-typescript",
 				"rules": {
+					"@typescript-eslint/no-misused-promises": "off",
+					"@typescript-eslint/require-await": "off",
```

Comment:
Buggy: it doesn't recognize returning a promise. It's already on top of some arguably annoying rules and it conflicts with them:

e.g.

```js
() => Promise.resolve()
  ✖  173:17  Functions that return promises must be async. @typescript-eslint/promise-function-async
```

```js
async () => Promise.resolve()
  ✖  178:17  Async arrow function has no await expression. @typescript-eslint/require-await
```

```js
async () => await Promise.resolve()
  ✖  173:29  Redundant use of await on a return value. no-return-await
```

The first one was perfectly fine.

Reply:
https://github.com/typescript-eslint/typescript-eslint/pull/674#issuecomment-526128992